### PR TITLE
support "ext.packaging.carvel.dev/fetch-%d-secret-name" on PackageInstall to set secret on underlying App CR

### PR DESCRIPTION
**Pull request by cppforlife**

### CODE REVIEW

File: pkg/packageinstall/app_test.go [hunk 1]
```diff
@@ -364,3 +364,88 @@ func TestAppManuallyControlled(t *testing.T) {
...
+		},
+	}
+
+	// Not interesting in metadata in this test
+	app.ObjectMeta = metav1.ObjectMeta{}
+
+	if !reflect.DeepEqual(expectedApp, app) {
+		bs, _ := yaml.Marshal(app)
+		t.Fatalf("App does not match expected app: (actual)\n%s", bs)
+	}
```
Comment (danielhelfand) [inline pkg/packageinstall/app_test.go]: Let's try and use the new testify library here to provide a better diff.

File: pkg/packageinstall/app_test.go [hunk 2]
```diff
@@ -364,3 +364,88 @@ func TestAppManuallyControlled(t *testing.T) {
...
+					},
+				},
+			},
+		},
+	}
+
+	app, err := packageinstall.NewApp(&kcv1alpha1.App{}, ipkg, pkgVersion)
+	if err != nil {
+		t.Fatalf("Expected no err, but was: %s", err)
+	}
```
Comment (danielhelfand) [inline pkg/packageinstall/app_test.go]: Let's try and use the new testify library here: `require.NotNil(t, err)`

**Comment by joe-kimmel-vmw** [comment]
I have some newbie questions about larger context:   is there a "user story" for "I need to associate secrets to specific fetch stages" that we currently aren't serving at all?   or in other words is there already a partial solution for a more common version of this end-user desire,  or is this going to be our answer for all such cases?

**Comment by cppforlife** [comment]
@joe-kimmel-vmw this is a quick&dirty way to expose users of PackageInstall to be able to set creds when installing a package. there is no way to do that today from the package consumer side (only from package author side since they can hard code some secret ref name). we may introduce first class support for specifying fetch secrets on PackageInstall in future versions.

**Comment by joe-kimmel-vmw** [comment]
> [...] from the package consumer side (only from package author side since [...]

cool,  thanks - i think ⬆️  is what i was missing: possibilities as a consumer vs. as an author.   

anyways agree it's a little on the hacky side but i can't think of a better proposal atm.   within that (and I do agree with @danielhelfand  that the testify libraries give better error messages),   LGTM

**Pull request merged**